index.bs

```diff
@@ -1203,6 +1224,73 @@ Gets the {{Error}} object passed to {{SensorErrorEventInit}}.
 </div>
 
 
+<h3 dfn>Find the reporting frequency of a Sensor</h3>
+
+<div algorithm="find the reporting frequency of a sensor">
+
+    : input
+    :: |sensor_instance|, a {{Sensor}} object.
+    : output
+    :: |frequency|, a [=frequency=].
+    1.  Let |frequency| be `null`.
+    1.  Let |f| be |sensor_instance|.{{[[desiredPollingFrequency]]}}.
+        1. if |f| is set,
+            1. set |frequency| to |f| capped by the upper and lower [=current polling frequency=]
+               bounds for the associated [=sensor=].
+        1.  Otherwise,
+            1. user agent can assign |frequency| to an appropriate value.
+    1. return |frequency|.
+</div>
+
+
+<h3 dfn>Report Latest Reading Updated</h3>
+
+<div algorithm="report latest reading updated">
+
+    : input
+    :: |sensor_instance|, a {{Sensor}} object.
+    : output
+    :: None
+
+    1.  If |sensor_instance|.{{[[readingUpdateRequested]]}} is `true`,
+        1. Abort these steps. 
+    1.  Set |sensor_instance|.{{[[readingUpdateRequested]]}} to `true`.
+    1.  Let |lastReportedTimestamp| be |sensor_instance|.{{[[reading]]}}["timestamp"].
+    1.  If |lastReportedTimestamp| is not set
+        1.  Invoke the [=Update Sensor Reading=] abstract operation, passing it |sensor_instance| as argument.
+        1.  Abort these steps.
+    1.  Let |reportingFrequency| be result of invoking the [=Find the reporting frequency of a Sensor=]
+        abstract operation.
+    1.  If |reportingFrequency| is not set
```

Comment:
`Find the reporting frequency of a Sensor` algorithm always returns something, so,  branch `If |reportingFrequency| is not set` should never be executed.